api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImplTest.java

```diff
@@ -72,21 +100,183 @@ public void shouldReturnNullWithUnknownEncounterUuid() {
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getPatient().getPerson().getPersonName().getFullName(),
+		    equalTo(PATIENT_FULL_NAME));
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersBySubjectFamilyName() {
+		ReferenceParam subjectReference = new ReferenceParam();
+		
+		subjectReference.setValue(PATIENT_FAMILY_NAME);
+		subjectReference.setChain(Patient.SP_FAMILY);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, null, subjectReference);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getPatient().getPerson().getPersonName().getFamilyName(),
+		    equalTo(PATIENT_FAMILY_NAME));
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersBySubjectGivenName() {
+		ReferenceParam subjectReference = new ReferenceParam();
+		
+		subjectReference.setValue(PATIENT_GIVEN_NAME);
+		subjectReference.setChain(Patient.SP_GIVEN);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, null, subjectReference);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getPatient().getPerson().getPersonName().getGivenName(),
+		    equalTo(PATIENT_GIVEN_NAME));
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersBySubjectIdentifier() {
+		ReferenceParam subjectReference = new ReferenceParam();
+		
+		subjectReference.setValue(PATIENT_IDENTIFIER);
+		subjectReference.setChain(Patient.SP_IDENTIFIER);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, null, subjectReference);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getPatient().getPatientIdentifier().getIdentifier(),
 		    equalTo(PATIENT_IDENTIFIER));
 	}
 	
 	@Test
-	public void findEncountersByWrongPatientIdentifier_shouldReturnEmptyCollectionOfEncounters() {
-		List<Encounter> encounters = dao.findEncountersByPatientIdentifier(WRONG_PATIENT_IDENTIFIER);
-		assertThat(encounters, notNullValue());
-		assertThat(encounters, is(empty()));
+	public void searchForEncounters_shouldReturnEmptyCollectionOfEncountersByWrongSubjectIdentifier() {
+		ReferenceParam subjectReference = new ReferenceParam();
+		
+		subjectReference.setValue(WRONG_PATIENT_IDENTIFIER);
+		subjectReference.setChain(Patient.SP_IDENTIFIER);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, null, subjectReference);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, is(empty()));
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersByParticipantIdentifier() {
+		ReferenceParam participantReference = new ReferenceParam();
+		
+		participantReference.setChain(Practitioner.SP_IDENTIFIER);
+		participantReference.setValue(PATIENT_IDENTIFIER);
```

Comment:
The constant we use here should be `PARTICIPANT_IDENTIFIER` even if it has the same value as `PATIENT_IDENTIFIER`.

Reply:
Also remember that the `chain` property needs to be set *after* the value.